lib/data/submission.js

```diff
@@ -43,6 +43,14 @@ const submissionXmlToFieldStream = (fields, xml, includeStructuralAttrs = false,
   const stack = new SchemaStack(fields, true);
   let textBuffer = ''; // agglomerates text nodes that come as multiple events.
   const parser = new hparser.Parser({
+    onend: () => {
+      if (!stack.hasExited()) {
+        outStream.destroy(new Error('Stack has not exited yet'));
+      } else {
+        parser.reset();
```

Comment:
I don't feel quite sure about what `parser.reset()` is doing here. I don't see `reset()` mentioned in the docs for htmlparser2 (though I know we're using an old version). If the parser has ended, and we're not reusing the parser, why do we need to reset it? Part of the reason I'm asking is that I wonder whether `reset()` had more of an effect before. Could `reset()` have had more of an effect when it was called in `onclosetag`, like causing the parser to stop parsing early?

Reply:
> I don't feel quite sure about what `parser.reset()` is doing here.

Copied from current line 70.  I'm also not sure why it's required, but my guess is similar to yours.  Would you rather remove it?